pkg/controller/podprobemarker/pod_probe_marker_controller.go

```diff
@@ -234,15 +236,47 @@ func (r *ReconcilePodProbeMarker) updateNodePodProbes(ppm *appsv1alpha1.PodProbe
 				exist = true
 				for j := range ppm.Spec.Probes {
 					probe := ppm.Spec.Probes[j]
+					if podProbe.IP == "" {
+						podProbe.IP = pod.Status.PodIP
+					}
+					if probe.Probe.TCPSocket != nil {
+						probe, err = convertTcpSocketProbeCheckPort(probe, pod)
+						if err != nil {
+							klog.Errorf("Failed to convert tcpSocket probe port, err: %v, pod: %v/%v", err, pod.Namespace, pod.Name)
+							continue
+						}
+					}
+					if probe.Probe.HTTPGet != nil {
+						probe, err = convertHttpGetProbeCheckPort(probe, pod)
+						if err != nil {
+							klog.Errorf("Failed to convert httpGet probe port, err: %v, pod: %v/%v", err, pod.Namespace, pod.Name)
+							continue
+						}
+					}
 					setPodContainerProbes(podProbe, probe, ppm.Name)
 				}
 				break
 			}
 		}
 		if !exist {
-			podProbe := appsv1alpha1.PodProbe{Name: pod.Name, Namespace: pod.Namespace, UID: string(pod.UID)}
+			podProbe := appsv1alpha1.PodProbe{Name: pod.Name, Namespace: pod.Namespace, UID: string(pod.UID), IP: pod.Status.PodIP}
 			for j := range ppm.Spec.Probes {
 				probe := ppm.Spec.Probes[j]
+				// look up a port in a container by name & convert container name port
+				if probe.Probe.TCPSocket != nil {
+					probe, err = convertTcpSocketProbeCheckPort(probe, pod)
+					if err != nil {
```

Comment:
plz add ut of Reconcile for tcpsocket and httpget

Reply:
Fixed